src/lib.rs

```diff
@@ -527,21 +535,13 @@ impl Section {
         // Patch all labels into the section's contents.
         let mut section = self;
         section.final_size.set_const(section.size());
-        let references: Vec<Reference> = section.references.to_vec();
-        let mut ok = true;
-        section = references.iter().cloned().fold(section, |s, r| {
-            if let Some(val) = r.label.value() {
-                s.store_label_value(val, r.offset, r.endian, r.size)
-            } else {
-                ok = false;
-                s
-            }
-        });
-        if ok {
-            Some(section.contents.into_inner())
-        } else {
-            None
+
+        for rf in section.references.clone() {
+            let val = rf.label.value()?;
+            section.store_label_value(val, rf.offset, rf.endian, rf.size);
```

Comment:
Hah, well, I think this change is pretty compelling evidence for why this is better. :)